torch/csrc/autograd/function.h

```diff
@@ -49,6 +51,59 @@ struct FunctionFlags {
   function_list next_functions;
 };
 
+namespace {
+
+// Why can't we just combine the set_variable and set_tensor variants
+// into one set of overloads?  The problem is Variable is convertible
+// to both Tensor and ArrayRef<Variable>, making the overload ambiguous.
```

Comment:
Variables are implicitly convertible to `ArrayRef<Variable>`?? What? Why?

Reply:
Yes.

```
    /// Construct an ArrayRef from a single element.
    /*implicit*/ ArrayRef(const T &OneElt)
      : Data(&OneElt), Length(1) {}
```

Inherited from LLVM. I guess it is useful sometimes, but it is very annoying for Tensor/Variable overloads.

Reply:
omg can we remove it? it makes no sense, if you want to pass an `ArrayRef` you can always do `{ arg }` to make one